packages/block-editor/src/utils/fit-text-frontend.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Frontend fit text functionality.
+ * Automatically detects and initializes fit text on blocks with the has-fit-text class.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { optimizeFitText } from './fit-text-utils';
+
+/**
+ * Counter for generating unique element IDs.
+ */
+let idCounter = 0;
+
+/**
+ * Get or create a unique style element for a fit text element.
+ *
+ * @param {string} elementId Unique identifier for the element.
+ * @return {HTMLElement} Style element.
+ */
+function getOrCreateStyleElement( elementId ) {
+	const styleId = `fit-text-${ elementId }`;
+	let styleElement = document.getElementById( styleId );
+	if ( ! styleElement ) {
+		styleElement = document.createElement( 'style' );
+		styleElement.id = styleId;
+		document.head.appendChild( styleElement );
+	}
+	return styleElement;
+}
+
+/**
+ * Generate a unique identifier for a fit text element.
+ *
+ * @param {HTMLElement} element The element to identify.
+ * @return {string} Unique identifier.
+ */
+function getElementIdentifier( element ) {
+	if ( ! element.dataset.fitTextId ) {
+		element.dataset.fitTextId = `fit-text-${ ++idCounter }`;
+	}
+	return element.dataset.fitTextId;
+}
+
+/**
+ * Initialize fit text functionality for a single element.
+ *
+ * @param {HTMLElement} element Element with fit text enabled.
+ */
+function initializeFitText( element ) {
+	const elementId = getElementIdentifier( element );
+
+	const applyFitText = () => {
+		const styleElement = getOrCreateStyleElement( elementId );
+		const elementSelector = `[data-fit-text-id=\"${ elementId }\"]`;
+
+		// Style management callback
+		const applyStylesFn = ( css ) => {
+			styleElement.textContent = css;
+		};
+
+		optimizeFitText( element, elementSelector, applyStylesFn );
+	};
+
+	// Initial sizing
+	applyFitText();
+
+	// Watch for parent container resize
+	if ( window.ResizeObserver && element.parentElement ) {
+		const resizeObserver = new window.ResizeObserver( applyFitText );
+		resizeObserver.observe( element.parentElement );
+	}
+}
+
+/**
+ * Initialize fit text on all elements with the has-fit-text class.
+ */
+function initializeAllFitText() {
+	const elements = document.querySelectorAll( '.has-fit-text' );
+	elements.forEach( initializeFitText );
+}
+
+document.addEventListener( 'DOMContentLoaded', initializeAllFitText );
```

Comment:
Why we need two events to init?

Reply:
The reason was that DOMContentLoaded fires when the HTML is parsed and DOM is ready, but before images/stylesheets/fonts finish loading, so it would provide a faster calculation but innacurate as when fonts and some styles are loaded the dimensions may change. The load one would fire when styles etc are loaded and would correct the calculation. The idea was to avoid a big difference between an initial size and the final size, but i did some tests and I did not noticed differences so I will just use window.addEventListener( 'load', initializeAllFitText ); to simplify things.